app/controllers/v1/users/friends_controller.rb

```diff
@@ -3,13 +3,18 @@
 class V1::Users::FriendsController < ApplicationController
   before_action :pundit_user
   before_action do
-    @page = set_page
+    @page = set_page.to_i
   end
 
   def index
```

Comment:
You can abstract some of the logic, ending up with something like:

```
if method?
  render :friends, status: :ok
else
  render json: { message: 'No more friends to show' }
end

private

def method?
  ...
end
```

making your controller's action smaller.

Reply:
i.e.: https://medium.com/selleo/essential-rubyonrails-patterns-part-1-service-objects-1af9f9573ca1

Reply:
Got it! I am actually using this `if @page <= max_page` line in another place (and I predict I will use it in a few more others). So i will take it further and put in `application_controller` or in a helper. Which of the two would be more appropriate?

Reply:
I have created a service class called `Pagination` that handles all the pagination related duties. Yeah I guess I should have used a gem in the first place. But I just wanted to also see how I could roll it out manually. Will probably use a gem in succeeding projects :)